### ISSUE TITLE
Node version requirements outdated? / Weird bug [doc issue?]

### ISSUE THREAD

**Issue by apparentorder** [homebridge/homebridge#2401]
### Expected Behavior
Pairing with Home app Just Works™ with any Node version >4.something (as documented), or at least unsupported versions throw an error.

### Current Behavior
- Pairing with Node 8.x: Doesn't work ("wrong" password)
- Pairing with Node 13.x: Works fine

So, I've spent several hours trying to figure out why pairing didn't work for me. The debug log kept complaining about the password, so did the Home app (iOS 13.3), with a minimal configuration / default password.

I have tried pretty much everything, changing port / username / password regularly, and even the "forget Wifi trick" (#1794) and similar stunts like Airplane Mode. But no luck. But also no error messages whatsoever, everything suggested there really is a password mismatch.

Finally I noticed that I had installed node8 instead of node "latest". After changing to node13, _the very same setup / configuration worked immediately_.

This seems extremely odd to me, as I would have expected errors, one way or another, when the Node version isn't properly supported or cryptographic functions are used "differently".

I assume debugging this would be extremely time-consuming and somewhat pointless -- but simply bumping the required Node version to 13 doesn't make sense either. This could even be FreeBSD- and/or ARM-specific behavior. Maybe a FAQ entry?

### Steps to Reproduce (for bugs)
I've been following this article: https://blog.zs64.net/2017/12/hooking-up-harmony-hub-and-generic-zigbee-lights-to-homekit/
Basically the steps are:
- pkg install npm-node8 avahi-libdns
- npm install mdns, homebridge (global, unsafe)
- run with minimalist/default config.json

### Versions

- **Homebridge**: 0.4.50
- **Node**:       13.3.0 [https://www.freshports.org/www/node]
- Node8: 8.16.2 [https://www.freshports.org/www/node8]
- **OS**:         FreeBSD 12.1-RELEASE-p1 on armv7 (RPi2)

### Homebridge Log / Command Output
Debug log gives:
```
Error while checking pincode: client did not use the same password
```

**Comment by jvmahon** [comment]
hap-nodejs, the underlying library, recently updated their minimum node requirement to ">=10.17.0" so I think it would make sense if homebridge used the same minimum.

**Comment by apparentorder** [comment]
fwiw, that makes good sense to me :-) I'd be happy to re-test and verify that it still works with 10.17.0 >= x < 11, if that would help.

**Comment by apparentorder** [comment]
> If this issue was overlooked, forgotten, or should remain open for any other reason, please reply here

As far as I can tell, it's still relevant: The documentation still says 4.x. And if there is indeed a requirement to have >= 10.17.0 for dependencies, as @jvmahon said, a version check on startup might make sense -- so running an unsupported version becomes immediately obvious to the user.

**Comment by oznu** [comment]
We are bumping this to 10+ in the next release (currently in beta).

You can see the beta branch has the readme updated to reflect this.

**Comment by speg** [comment]
I just had a similar problem, updating from v10.15.2 to v14.16.0 fixed it for me.
